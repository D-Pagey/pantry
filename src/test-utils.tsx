import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
