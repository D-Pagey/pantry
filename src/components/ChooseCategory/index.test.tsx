import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../test-utils';

import { ChooseCategory, ChooseCategoryProps } from '.';

const props: ChooseCategoryProps = {
    handleClick: () => null
};

describe('ChooseCategory component', () => {
    it('should render', () => {
        const { container } = render(<ChooseCategory {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onClick with category', () => {
        const handleClick = jest.fn();

        const { getByText } = render(<ChooseCategory {...props} handleClick={handleClick} />);

        userEvent.click(getByText('Meat'));

        expect(handleClick).toHaveBeenCalledWith('meat');
    });
});
