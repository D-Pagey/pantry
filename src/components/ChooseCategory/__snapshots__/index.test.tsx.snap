// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCategory component should render 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border: 1px solid rgba(52,136,65,0.6);
  border-radius: 16px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 115px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease 0s;
  transition: all 0.3s ease 0s;
  width: 144px;
}

.c3:hover {
  background-color: #348841;
}

.c4 {
  color: #348841;
  font-size: 18px;
}

.c2:hover .c4 {
  color: white;
}

.c0 {
  margin: 0;
  padding: 0 0 1.5rem;
  font-weight: normal;
}

.c1 {
  display: grid;
  grid-gap: 24px;
  grid-template-columns: 1fr 1fr;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

<div>
  <h3
    class="c0"
  >
    What type of food?
  </h3>
  <div
    class="c1"
  >
    <button
      class="c2 c3"
      data-testid="categoryCardvegetables"
      type="button"
    >
      <p
        class="c4"
      >
        Vegetables
      </p>
    </button>
    <button
      class="c2 c3"
      data-testid="categoryCardfruit"
      type="button"
    >
      <p
        class="c4"
      >
        Fruit
      </p>
    </button>
    <button
      class="c2 c3"
      data-testid="categoryCarddairy"
      type="button"
    >
      <p
        class="c4"
      >
        Dairy
      </p>
    </button>
    <button
      class="c2 c3"
      data-testid="categoryCardmeat"
      type="button"
    >
      <p
        class="c4"
      >
        Meat
      </p>
    </button>
    <button
      class="c2 c3"
      data-testid="categoryCardfish"
      type="button"
    >
      <p
        class="c4"
      >
        Fish
      </p>
    </button>
    <button
      class="c2 c3"
      data-testid="categoryCard[opened tin]"
      type="button"
    >
      <p
        class="c4"
      >
        [Opened Tin]
      </p>
    </button>
  </div>
</div>
`;
